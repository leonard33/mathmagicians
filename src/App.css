/* stylelint-disable keyframes-name-pattern */
/* stylelint-disable selector-class-pattern */
