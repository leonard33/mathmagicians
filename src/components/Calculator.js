/* eslint-disable react/button-has-type */
import React, { Component } from 'react';
import './Calculate.css';

const createDigit = () => {
  const digits = [];
  for (let i = 9; i >= 1; i -= 1) {
    digits.push(<button key={i}>{i}</button>);
  }
  return digits;
};
// eslint-disable-next-line react/prefer-stateless-function
export default class Calculator extends Component {
  render() {
    return (
      <div className="calculate">
        <div className="display"><span>0</span></div>
        <div className="calc-digits">
          <div className="calc-left">
            <div className="operations-top">
              <button>AC</button>
              <button>+/-</button>
              <button>%</button>
            </div>
            <div className="digits">
                {createDigit()}
                <button>0</button>
                <button>.</button>
              </div>
          </div>
          <div className="calc-right">
            <div className="operations-right">
              <button>/</button>
              <button>*</button>
              <button>-</button>
              <button>=</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
