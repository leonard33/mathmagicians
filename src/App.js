import React, { Component } from 'react';
import Calculator from './components/Calculator';
import './App.css';
import { Routes, Route } from "react-router-dom";
import Quote from './components/Quote';
import Home from './components/Home';
import Nav from './components/Nav';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="App">
        <Nav/>
        <Routes>
         <Route path="/" element={<Home />} />
         <Route path="quotes" element={<Quote />} />
         <Route path="calculator" element={<Calculator />} />
        </Routes>
      </div>
    );
  }
}
/*
import Calculator from './components/Calculator';
import './App.css';

function App() {
  return (
    <div className="App">
      <Calculator />
    </div>
  );
}

export default App; */
