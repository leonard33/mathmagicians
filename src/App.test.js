/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
// import { cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';
import { render, screen } from '@testing-library/react';
import Calculator from './components/Calculator';
import userEvent from '@testing-library/user-event';
import operate from './logic/operate';


// afterEach(cleanup)

test('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter>
      <App />
    </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


test("testing the title", () => {
  render(<Calculator/>);
  const title = screen.getByTestId("mytitle");
  expect(title).toBeInTheDocument()
});

// Test links on click to the page component
// test calculate link
describe('when you click the calculate link', () => {
  it('should render Calculate page content', () => {
    render(
      <BrowserRouter>
      <App/>
      </BrowserRouter>
    )
    userEvent.click(screen.getByRole('link', {  name: /calcultor/i}));
    expect(screen.getByRole('heading', {  name: /lets do some math/i})).toBeInTheDocument();
  })
})

// test Home link
describe('When you click the Home link', () => {
  it('should rende Home page content', () => {
    render(
      <BrowserRouter>
      <App/>
      </BrowserRouter>
    )
    userEvent.click(screen.getByRole('link', {  name: /home/i}));
    expect(screen.getByText(/welcome to our page !/i)).toBeInTheDocument();
  })
})

// Test Quote link
describe('when you click Quote link', () => {
  it('should render Quote page content', () => {
    render(
      <BrowserRouter>
      <App/>
      </BrowserRouter>
    )
    userEvent.click(screen.getByRole('link', {  name: /quotes/i}));
    // eslint-disable-next-line no-useless-escape
    expect(screen.getByText(  /mathematics is not about numbers equations, computations or algorithms it is about understanding \-william paul thurston/i  )).toBeInTheDocument();
  })
})

// Test for calculator operation ("+", "-", "x", "รท", "%")
describe('unit test operations', () => {
  test('5 + 5 = 10', () => {
    const operation = operate(5, 5, '+');
    expect(operation).toBe('10')
  })

  test('50 - 8 = 42', () => {
    const operation = operate(50, 8, '-');
    expect(operation).toBe('42');
  })

  test('2.5 x 0.005 = 0.0125', () => {
    const operation = operate(2.5, 0.005, 'x');
    expect(operation).toBe('0.0125')
  })

  test('5 รท 2 = 2.5', () => {
    const operation = operate(5, 2, 'รท');
    expect(operation).toBe('2.5')
  })

  test('78 % 5 = 3', () => {
    const operation = operate(78, 5, '%');
    expect(operation).toBe('3')
  })
})