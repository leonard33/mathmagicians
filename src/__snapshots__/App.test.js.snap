// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="logo"
    >
      Math Magician
    </div>
    <ul
      data-testid="mylist"
    >
      <a
        className="lang"
        href="/"
        onClick={[Function]}
      >
        <li>
          Home
        </li>
      </a>
      <a
        className="lang"
        href="/calculator"
        onClick={[Function]}
      >
        <li>
          Calcultor
        </li>
      </a>
      <a
        className="lang"
        href="/quotes"
        onClick={[Function]}
      >
        <li>
          Quotes
        </li>
      </a>
    </ul>
  </div>
  <div>
    Welcome to our page !
  </div>
</div>
`;
